src/widgets/algorithm_setup_widget.py

```diff
@@ -14,42 +17,94 @@ def __init__(self, parent, engine):
+        self.first_column.addStretch(2)
+        self.first_column.addWidget(self.technique_group)
+        self.first_column.addStretch(1)
+        self.first_column.addWidget(self.algorithm_selection_group)
+        self.first_column.addStretch(2)
 
         # options group
-        self.options_group = QGroupBox(self.frame)
-        self.options_group.setGeometry(QRect(340, 100, 281, 251))
+        self.options_group = QGroupBox()
         self.options_group.setTitle("Options")
+        self.options_group.setMinimumSize(220, 200)
+        self.options_group_layout = QFormLayout(self.options_group)
+
+        self.num_clusters_spinbox = QSpinBox()
+        self.num_clusters_spinbox.setMinimum(2)
+        self.num_clusters_spinbox.setMaximum(engine.get_maximum_clusters())
+        self.num_clusters_spinbox.setValue(5)
+        self.options_group_layout.addRow(QLabel("Number of clusters:"), self.num_clusters_spinbox)
+
+        self.metrics_spinbox = QSpinBox()
+        self.metrics_spinbox.setMinimum(1)
+        self.metrics_spinbox.setValue(1)
+        self.metrics_spinbox.setMaximum(6)
+        self.options_group_layout.addRow(QLabel("Exponent in metrics:"), self.metrics_spinbox)
+
+        self.second_column.addStretch()
+        self.second_column.addWidget(self.options_group)
+        self.second_column.addStretch()
+
+        self.vertical_layout.addStretch(2)
+        self.vertical_layout.addLayout(self.first_column)
+        self.vertical_layout.addStretch(1)
+        self.vertical_layout.addLayout(self.second_column)
+        self.vertical_layout.addStretch(2)
 
-        self.num_clusters_spinbox = QSpinBox(self.options_group)
-        self.num_clusters_spinbox.setGeometry(QRect(160, 70, 81, 31))
+        # button
+        self.run_button = QPushButton(self.frame)
+        self.run_button.setText("Submit and run")
+        self.run_button.setFixedWidth(300)
+        self.run_button.clicked.connect(partial(self.click_listener, 'run'))
 
-        self.num_clusters_label = QLabel(self.options_group)
-        self.num_clusters_label.setGeometry(QRect(20, 60, 131, 51))
-        self.num_clusters_label.setAlignment(Qt.AlignCenter)
-        self.num_clusters_label.setText("number of clusters")
+        self.layout.addStretch()
+        self.layout.addLayout(self.vertical_layout)
+        self.layout.addWidget(self.run_button, alignment=Qt.AlignCenter)
+        self.layout.addStretch()
 
-        self.num_clusters_label_2 = QLabel(self.options_group)
-        self.num_clusters_label_2.setGeometry(QRect(20, 160, 131, 51))
-        self.num_clusters_label_2.setAlignment(Qt.AlignCenter)
-        self.num_clusters_label_2.setText("initialization method")
+        # dialog error
+        self.dialog = QDialog(self)
+        self.dialog.setFixedHeight(50)
+        self.dialog.setFixedWidth(400)
+        self.dialog.setWindowTitle("Error message")
+        QLabel("You didn't set data.\nPlease return to 'IMPORT DATA' window.", self.dialog)
```

Comment:
```suggestion
        QLabel("Data not set.\nPlease return to the 'IMPORT DATA' window.", self.dialog)
```
\+ turn that into QMessageBox to match what Tomek did in PreprocessingWidget::get_data